lib/mp-readline/readline.c

```diff
@@ -98,6 +99,38 @@ typedef struct _readline_t {
 
 STATIC readline_t rl;
 
+size_t get_previous_word_cursor_pos() {
+  size_t cursor_pos = rl.cursor_pos;
+  bool in_leading_nonalphanum = true;
+  while (cursor_pos > 0) {
+    if (!isalnum((unsigned char) rl.line->buf[cursor_pos - 1])) {
+      if (!in_leading_nonalphanum) {
+        break;
+      }
+    } else if (in_leading_nonalphanum) {
+      in_leading_nonalphanum = false;
+    }
+    cursor_pos -= 1;
+  }
+  return cursor_pos;
+}
+
+size_t get_next_word_cursor_pos() {
```

Comment:
Would be good to combine this function with the one above, parameterised by a "direction" argument, to reduce code size.

Reply:
Combined into single [`size_t get_word_cursor_pos(bool direction)`](https://github.com/micropython/micropython/pull/5024/commits/00b2a0e2609ed84df5be05af728c551c457e622f#diff-ec323dc5c7454143942798519678d424R105)